ReactNativeClient/lib/shim-init-node.js

```diff
@@ -445,6 +455,33 @@ function shimInit() {
 	shim.pathRelativeToCwd = (path) => {
 		return toRelative(process.cwd(), path);
 	};
+
+	function isKeytarAvailable() {
+		return shim.isElectron() && !shim.isPortable() && keytar;
+	}
+
+	shim.loadSecureItems = async (appId) => {
+		if (isKeytarAvailable()) {
+			const secureItems = await keytar.findCredentials(appId);
+			// secureItems is an array of { account: 'foo', password: 'bar' }
+			// but it should be an array of { key: 'foo', value: 'bar' }
+			// https://stackoverflow.com/a/50101979
+			for (const item of secureItems) {
```

Comment:
No mutation, so please create a new array `newSecureItems` and then `newSecureItems.push({ key: .... value: ....})`

Reply:
I've done this. But, why is mutation not preferred? I'm trying to learn :grimacing:

Reply:
In this particular case, we don't really know how `findCredentials` works internally - probably it returns a new array every time you call it. Probably. But it could also for example cache the data and return the same array every time. Then if you modify that array, you are changing data used by keytar and could break something.

I don't think it would actually happen, but to be safe it always better not to mutate data, especially data you didn't create yourself. It's mostly defensive programming, because object ownership it not well defined in JavaScript.

Reply:
Thanks for explaining!